src/librustc/ty/mod.rs

```diff
@@ -1981,10 +1981,30 @@ impl<'a, 'gcx, 'tcx> AdtDef {
     }
 
     #[inline]
-    pub fn is_non_exhaustive(&self) -> bool {
+    fn is_non_exhaustive(&self) -> bool {
         self.flags.intersects(AdtFlags::IS_NON_EXHAUSTIVE)
     }
 
+    #[inline]
+    pub fn is_enum_non_exhaustive(&self) -> bool {
+        match self.adt_kind() {
+            AdtKind::Enum => self.is_non_exhaustive(),
+            AdtKind::Struct | AdtKind::Union => {
+                bug!("is_non_exhaustive_enum called on non-enum `{:?}`", self);
+            }
+        }
+    }
+
+    #[inline]
+    pub fn is_univariant_non_exhaustive(&self) -> bool {
```

Comment:
I think we should call this method:

```rust
can_extend_field_list
```

and -- really -- I think it should be on the variant. But if we want to keep it here, perhaps a name like `can_extend_univariant_field_list` (again with a comment linking to how it maps to Rust source).

Reply:
I like to use the name `nonexhaustive` here, because that's how the feature is called.